packages/messaging/android/src/main/java/io/invertase/firebase/messaging/ReactNativeFirebaseMessagingStoreImpl.java

```diff
@@ -28,18 +28,21 @@ public void storeFirebaseMessage(RemoteMessage remoteMessage) {
           reactToJSON(remoteMessageToWritableMap(remoteMessage)).toString();
       //      Log.d("storeFirebaseMessage", remoteMessageString);
       UniversalFirebasePreferences preferences = UniversalFirebasePreferences.getSharedInstance();
-      preferences.setStringValue(remoteMessage.getMessageId(), remoteMessageString);
-      // save new notification id
-      String notificationIds = preferences.getStringValue(S_KEY_ALL_NOTIFICATION_IDS, "");
-      notificationIds += remoteMessage.getMessageId() + DELIMITER; // append to last
-      preferences.setStringValue(S_KEY_ALL_NOTIFICATION_IDS, notificationIds);
 
-      // check and remove old notifications message
+      // remove old notifications message before store to free space as needed
```

Comment:
the diff for the 5th commit is easier to understand in split view, it's mostly code motion swapping the two chunks of logic, not code change